apps/web/app/(app)/mail/page.tsx

```diff
@@ -68,17 +74,50 @@ export default function Mail({
     setRefetchEmailList({ refetch });
   }, [refetch, setRefetchEmailList]);
 
+  useEffect(() => {
+    if (data) {
+      setAllThreads(data.threads);
+      setNextPageTokenId(data.nextPageToken);
+    }
+  }, [data]);
+
+  // Fetch new list of results based on nextPageToken when load more is clicked
+  const handleLoadMoreThreads = async () => {
+    if (!nextPageTokenId) return;
+    setIsLoadMoreLoading(true);
+    mutate(
+      async (currentData) => {
+        const res = await fetch(
+          `/api/google/threads?nextPageToken=${nextPageTokenId}&${new URLSearchParams(
+            query as Record<string, string>,
+          ).toString()}`,
+        );
+        const newData: ThreadsResponse = await res.json();
+
+        // update the threads and nextPageToken
+        return {
+          threads: [...(currentData?.threads || []), ...newData.threads],
+          nextPageToken: newData.nextPageToken,
+        };
+      },
+      { revalidate: false },
+    ).finally(() => setIsLoadMoreLoading(false));
```

Comment:
Not critical but use try/await/catch instead of finally to be more consistent with the rest of the codebase.

Reply:
No need of this anymore!